tests/dialects/test_seq.py

```diff
@@ -22,3 +25,15 @@ def test_clockdivider_verify():
         match="Operation does not verify: pow2 has to be an 8-bit signless integer",
     ):
         clock_div.verify()
+
+
+def test_compreg_builder():
+    data_val = TestSSAValue(IntegerType(5))
+    bool_val = TestSSAValue(i1)
+    clock_val = TestSSAValue(clock)
+
+    CompRegOp(data_val, clock_val).verify()
+
+    with_reset = CompRegOp(data_val, clock_val, (bool_val, data_val))
+    with_reset.verify()
+    assert with_reset.reset is not None
```

Comment:
This test case would benefit from a short explanation what it is actually testing